enclave_wrangler/main.py

```diff
@@ -46,19 +49,43 @@ def _datetime_palantir_format() -> str:
     return datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.%fZ")[:-4] + 'Z'
 
 
-def run(input_csv_folder_path):
-    """Main function"""
+def _load_standardized_input_df(
+    path, integer_id_fields=['enclave_codeset_id', 'codeset_id', 'internal_id']) -> pd.DataFrame:
+    """Loads known input dataframe in a standardized way:
+        - Sets data types
+        - Fill na values"""
+    df: pd.DataFrame = pd.read_csv(path).fillna('')
+
+    # Strip: remove the spaces in the beginning and the end of the name
+    df.columns.str.lstrip()
+    df.columns.str.rstrip()
+
+    # Float->Int: For some reason, was being read with .0's at the end.
+    for field in [x for x in integer_id_fields if x in list(df.columns)]:
+        df[field] = pd.to_numeric(df[field], errors='coerce')\
+            .astype('Int64')
+
+    return df
+
+
+def load_cache_if_valid(input_csv_folder_path) -> Union[pd.DataFrame, None]:
+    """Checks `enclave_codeset_id` and, if no empty vals, uses this file as cache."""
+    df: pd.DataFrame = _load_standardized_input_df(os.path.join(input_csv_folder_path, 'code_sets.csv'))
+    ids = list(df['enclave_codeset_id'])
+    valid = not any([x == '' for x in ids])
+
+    return df if valid else None
+
+
+def post_to_enclave_and_update_code_sets_csv(input_csv_folder_path) -> pd.DataFrame:
```

Comment:
Abstracted this into its own function mainly fo the purposes of working with caching.